aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -511,6 +518,22 @@ module aptos_framework::stake {
         let amount = coin::value<AptosCoin>(&coins);
         assert!(amount > 0, error::invalid_argument(EINVALID_STAKE_AMOUNT));
 
+        // Only track and validate voting power increase for active and pending_active validator.
+        // Pending_inactive validator will be removed from the validator set in the next epoch.
+        // Inactive validator's total stake will be tracked when they join the validator set.
+        let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
+        // Search directly rather using get_validator_state to save on unnecessary loops.
+        if (option::is_some(&find_validator(&validator_set.active_validators, pool_address)) ||
```

Comment:
what I had in mind is something like this, but probably not for this PR
```
  fun is_active_validator(pool_address: address): bool {
    if !exist<ValidatorConfig>(pool_address) return false;
    let validator_index = borrow_global<ValidatorConfig>(pool_address).validator_index;
    let active_validators = &borrow_global<ValidatorSet>(@aptos_framework).active_validators;
    if validator_index >= vector::length(active_validators) return false;
    return vector::borrow(&active_validators, validator_index) == pool_address
  }
```

Reply:
Yeah here I need to check if they're either active or pending_active

Reply:
we can integrate this with `get_validator_state` (use this instead of find_validator for active_validators) in a followup PR, assuming pending vectors are small, this should be much better than before